MapboxNavigation/RouteVoiceController.swift

```diff
@@ -87,12 +87,14 @@ open class RouteVoiceController: NSObject, AVSpeechSynthesizerDelegate, AVAudioP
         NotificationCenter.default.addObserver(self, selector: #selector(alertLevelDidChange(notification:)), name: RouteControllerAlertLevelDidChange, object: nil)
         NotificationCenter.default.addObserver(self, selector: #selector(pauseSpeechAndPlayReroutingDing(notification:)), name: RouteControllerWillReroute, object: nil)
         NotificationCenter.default.addObserver(self, selector: #selector(didReroute(notification:)), name: RouteControllerDidReroute, object: nil)
+        NotificationCenter.default.addObserver(self, selector: #selector(voiceVolumeDidChange(notification:)), name: RouteVoiceControllerVoiceVolumeDidChange, object: nil)
     }
     
     func suspendNotifications() {
         NotificationCenter.default.removeObserver(self, name: RouteControllerAlertLevelDidChange, object: nil)
         NotificationCenter.default.removeObserver(self, name: RouteControllerWillReroute, object: nil)
         NotificationCenter.default.removeObserver(self, name: RouteControllerDidReroute, object: nil)
+        NotificationCenter.default.removeObserver(self, name: RouteVoiceControllerVoiceVolumeDidChange, object: nil)
```

Comment:
No need to post and observe a custom notification; instead:

* The toggle in MapboxNavigation should set the `MBNavigationVoiceVolume` user default (named with a prefix in order to avoid collisions with user defaults set in application code).
* Core Navigation should observe for `NSNotification.Name.didChangeNotification` and handle volume changes based on the `MBNavigationVoiceVolume` user default changing.
  * Alternatively, use KVO to observe for changes to `MBNavigationVoiceVolume` specifically.

Reply:
@1ec5 I followed your lead [here](https://github.com/mapbox/mapbox-navigation-ios/pull/673/commits/09bb80f10019faa59bf8c8c2d9fed62066200ac8), but for some reason it fails to enter navigation when this [line](https://github.com/mapbox/mapbox-navigation-ios/pull/673/commits/09bb80f10019faa59bf8c8c2d9fed62066200ac8#diff-eaaa217c63f0df828e942d653c6994ceR137) is present. Really strange.

Reply:
`NavigationSettings.voiceVolume`’s setter might need to synchronize the user defaults after setting the default, which it should probably do anyways.

Reply:
Here’s where the map SDK observes changes to user defaults:

https://github.com/mapbox/mapbox-gl-native/blob/d913f4be79f2970c0ba95c297816632241eb374b/platform/ios/src/MGLMapboxEvents.m#L206

Reply:
We don't have to worry about collisions, our defaults will be stored in a different plist and not the standard. This might be the reason why observing standard defaults doesn't trigger KVO.

https://github.com/mapbox/mapbox-navigation-ios/blob/b14107790a7647496e597590846a6c7ec77817cb/MapboxNavigation/Settings.swift#L18

Reply:
Ah, yes, that would do it. Do we need to store the setting in a separate plist? It’s not uncommon for libraries to use the standard user defaults but prefix each key to avoid collisions.